runtime/dom-set-context.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * @license
+ * Copyright (c) 2017 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+'use strict';
+
+import {assert} from '../platform/assert-web.js';
+import Template from '../shell/components/xen/xen-template.js';
+import {DomContext} from './dom-context.js';
+
+export class DomSetContext {
```

Comment:
What about calling this `DomCollectionContext`? I spent some time reviewing to determine whether it's a variant of `DomContext` that allows setting the context in some alternate manner before concluding that, no, it's `set` in that it handles a set of slots identifiable by subid. We seem to refer to sets as collections in most of runtime.

Or fit `subid` in the class name somehow?

Am fine as is but thought to raise for consideration in any case. Could also be worth adding brief class header comment here and in `DomContext` to explain them.

Reply:
+1 for `DomCollectionContext`

Reply:
I added a comment for now. I prefer to keep it DomSetContext for now, partly because DomCollectionContext is so long, but mainly because it is consistent with the "set slot" terminology we use in the manifests.

I'll merge it as is for now, because I have an unrelated follow up PR, but happy to rename later, if either of you feel strongly about it.